sql/mozfun/json/extract_string_map/metadata.yaml

```
@@ -0,0 +1,5 @@
+---
+friendly_name: JSON Extract String Map
+description: >-
+  Returns an array of key/value structs from a string representing a JSON map.
+  Non-string non-null values are encoded as json.
```

This is the first JS UDF in mozfun other than glam-specific UDFs. Minimally, I'd like the description here to specify that this is a JS UDF and that there are performance implications. Perhaps we should somehow encode the fact that this is JS in the name of the function or namespace.

Alternately, if the VPN use case is simple, we may be able to make some assumptions about structure and achieve what we need using some string substitutions without leaving SQL.

---

> Minimally, I'd like the description here to specify that this is a JS UDF and that there are performance implications.

```suggestion
  BigQuery Standard SQL JSON functions are insufficient to implement this function, so JS is being used and it may not perform well with large or numerous inputs.
  Non-string non-null values are encoded as json.
```

>  Perhaps we should somehow encode the fact that this is JS in the name of the function or namespace.

I think it would be appropriate and effective to rename this to `mozfun.json.js_extract_string_map`.

> Alternately, if the VPN use case is simple, we may be able to make some assumptions about structure and achieve what we need using some string substitutions without leaving SQL.

I'm pretty sure the VPN data isn't so kind as to disallow `"` and `,` within the recorded header values.